### ISSUE TITLE
Attribute initialization with clones

### ISSUE THREAD

**Issue by jemole** [ucsb-cs-education/hairball#7]
Hi!

When using clones it is quite common that attribute initialization is performed under "When I start as a clone" block, instead of under "When green flag clicked". However, this situation would not considered as INITIALIZED by the plug-in.

By modifying partition_scripts to receive two start_types this issue could be easily solved:

```python
def partition_scripts(scripts, start_type1, start_type2):
    ...
    if (HairballPlugin.script_start_type(script) == start_type1 or 
        HairballPlugin.script_start_type(script) == start_type2):
            match.append(script)
    ....

def attribute_state(cls, scripts, attribute):
    ...
    green_flag, other = partition_scripts(scripts, cls.HAT_GREEN_FLAG, cls.HAT_CLONE)
```

Best regards.

**Comment by bboe** [comment]
Would you like to make a pull request for it?

**Comment by bboe** [comment]
Resolved via #8.
